src/main/scala/mesosphere/marathon/state/Group.scala

```diff
@@ -254,19 +254,19 @@ object Group {
   private def validPorts: Validator[Group] = {
     new Validator[Group] {
       override def apply(group: Group): Result = {
-        val groupViolations = group.apps.flatMap { app =>
-          val ruleViolations = app.container.flatMap(_.servicePorts).toSeq.flatMap { servicePorts =>
-            for {
-              existingApp <- group.transitiveApps.toList
-              if existingApp.id != app.id // in case of an update, do not compare the app against itself
-              existingServicePort <- existingApp.container.flatMap(_.portMappings).toList.flatten.map(_.servicePort)
-              if existingServicePort != 0 // ignore zero ports, which will be chosen at random
-              if servicePorts contains existingServicePort
-            } yield RuleViolation(
-              app.id,
-              s"Requested service port $existingServicePort conflicts with a service port in app ${existingApp.id}",
-              None)
-          }
+        val groupViolations = group.transitiveApps.flatMap { app =>
+          val servicePorts = app.servicePorts.toSet
```

Comment:
The old code was actually buggy and wouldn't validate service ports defined through `portDefinitions`. This change fixes that.